datalab/datalab_session/data_operations/input_data_handler.py

```diff
@@ -27,15 +29,27 @@ def __init__(self, basename: str, source: str = None) -> None:
     self.source = source
     self.exit_stack = ExitStack()
     self.fits_file = self.exit_stack.enter_context(get_fits(basename, source))
-    self.sci_data = get_hdu(self.fits_file, 'SCI').data
+    self.sci_hdu = get_hdu(self.fits_file, 'SCI')
+    self.sci_data = self.sci_hdu.data
   
   def __del__(self):
     self.exit_stack.close()
 
+  def __enter__(self):
+    return self
+
+  def __exit__(self, exc_type, exc_val, exc_tb):
```

Comment:
Allowing this class to be used as a context manager in some cases, and explicitly doing a garbage collection when the file and data exits context. This is for those operations that operate on files sequentially like subtraction and normalization so we should be able to support infinite files without running out of memory.